pdp/target.go

```diff
@@ -192,7 +192,16 @@ type twoArgumentsFunctionType func(first, second Expression) Expression
 var TargetCompatibleExpressions = map[string]map[int]map[int]twoArgumentsFunctionType{
 	"equal": {
 		TypeString: {
-			TypeString: makeFunctionStringEqual}},
+			TypeString: makeFunctionStringEqual},
+		TypeInteger: {
+			TypeInteger: makeFunctionIntegerEqual},
+		TypeFloat: {
```

Comment:
Actually, here you missed float+int and int+float -> makeFunctionFloatEqual cases if you want to follow implicit conversion. However, I don't think it's worth to make it.

Reply:
Sounds good!

Actually, I thought about that. The options are:
- Promote the int to float and then compare as floats. This typically is not reliable. It might makes sense for greater than, but not for equal, IMHO.
- Cast/Round the float to int and then compare as int. This is potentially very confusing for the users.